infra/resources/_modules/key_vaults/io_com_key_vault.tf

```
@@ -0,0 +1,28 @@
+resource "azurerm_key_vault" "io_com_kv" {
+  name                     = "io-p-itn-com-kv"
```

this name does not adhere to our [naming convention](https://pagopa.github.io/dx/docs/conventions/azure-naming-convention/), please use the `naming_convention` function from the `dx` provider

```suggestion
  name = provider::dx::resource_name({
      prefix          = "io"
      name            = "com",
      domain          = "",
      resource_type   = "key_vault",
      environment     = "p",
      location        = "italynorth"
      instance_number = 1
  })
```

replace the hardcoded values with variables (you can declare a variable called `environment` similar to the one used in `web_apps` module)

to use `provider::dx::resource_name` you have to register the `pagopa-dx/azure` provider to `required_providers` in `main.tf`

```tf
dx = {
  source  = "pagopa-dx/azure"
  version = ">= 0.0.6, < 1.0.0"
}
```

---

[fa45699](https://github.com/pagopa/io-messages/pull/260/commits/fa4569988da8edd074bb893daaadd43516dc1a26)